src/test/java/core/basesyntax/DataHandlerStrategyTest.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.DataHandlerStrategy;
+import core.basesyntax.strategy.DataHandlerStrategyImpl;
+import core.basesyntax.strategy.handlers.BalanceDataHandler;
+import core.basesyntax.strategy.handlers.DataHandler;
+import core.basesyntax.strategy.handlers.PurchaseDataHandler;
+import core.basesyntax.strategy.handlers.ReturnDataHandler;
+import core.basesyntax.strategy.handlers.SupplyDataHandler;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class DataHandlerStrategyTest {
+    private static DataHandlerStrategy dataHandlerStrategy;
+    private static Map<FruitTransaction.Operation, DataHandler> enumHandlerMap;
+
+    @BeforeAll
+    static void setHandlerMap() {
+        enumHandlerMap = new HashMap<>();
+        DataHandler balance = new BalanceDataHandler();
+        enumHandlerMap.put(FruitTransaction.Operation.BALANCE, balance);
+        DataHandler returns = new ReturnDataHandler();
+        enumHandlerMap.put(FruitTransaction.Operation.RETURN, returns);
+        DataHandler purchase = new PurchaseDataHandler();
+        enumHandlerMap.put(FruitTransaction.Operation.PURCHASE, purchase);
+        DataHandler supply = new SupplyDataHandler();
+        enumHandlerMap.put(FruitTransaction.Operation.SUPPLY, supply);
+        dataHandlerStrategy = new DataHandlerStrategyImpl(enumHandlerMap);
```

Comment:
<img width="950" alt="image" src="https://github.com/mate-academy/jv-fruit-shop-tests/assets/32339836/395e3988-87cf-4024-aef0-9bf6816929c4">

Reply:
Not resolved